client_cli/src/main.rs

```diff
@@ -1060,6 +1062,196 @@ mod peer {
     }
 }
 
+mod param {
+    use super::*;
+    use clap::Subcommand;
+    use core::num::NonZeroU64;
+    use iroha::client::{self, *};
+    use iroha_data_model::isi::SetParameter;
+    use iroha_data_model::parameter::{self, *};
+
+    // #[derive(Parser, Debug)]
+    // #[command(author, version, about, long_about = None)]
+    #[derive(Subcommand, Debug)]
+    pub enum Args {
+        // #[command(subcommand)]
+        // Parameters(ParamCmd),
+        #[command(subcommand)]
+        Get(Box<Get>),
+        #[command(subcommand)]
+        Set(Box<Set>),
+    }
+
+    impl RunArgs for Args {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            match_all!((self, context), { Args::Get, Args::Set, })
+        }
+    }
+
+    #[derive(Subcommand, Debug)]
+    pub enum Get {
+        All,
+        #[command(flatten)]
+        Inner(ParamCmd),
+    }
+
+    /// Get values of parameters
+    impl RunArgs for Get {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            let client = context.client_from_config();
+            let vec = match self {
+                Self::All => client
+                    .request(client::parameter::all())
+                    .wrap_err("Failed to get all the parameters."),
+                Self::Inner(param) => client
+                    .build_query(client::parameter::all())
+                    .with_filter(ParamCmd::Sumeragi)
+                    .execute()
+                    .wrap_err("Failed to get the parameter value"),
+            }?;
+            context.print_data(&vec.collect::<QueryResult<Vec<_>>>()?)?;
+            Ok(())
+        }
+    }
+
+    /// Set values for parameters
+    #[derive(clap::Subcommand, Debug)]
+    pub enum Set {
+        All,
+        #[command(flatten)]
+        Inner(ParamCmd),
+    }
+
+    impl RunArgs for Set {
+        fn run(self, context: &mut dyn RunContext) -> Result<()> {
+            let set_param_value = SetParameter::new(0);
+	    // Not sure what to do.
+	    Ok(())
+        }
+    }
```

Comment:
And similarly, how do we exactly take the parameters from the hierarchy we have and actually set them? I believe we need to use `SetParameter` from `data_model::parameter`, but don't know how to go further on it.

Reply:
`SetParameter` is an `Instruction`, which can be submitted to an Iroha `Client`. You need to turn `self` into a `SetParameter` here, and then call `submit` (defined in this file).